weblate/api/serializers.py

```diff
@@ -127,6 +143,17 @@ class Meta:
             "url": {"view_name": "api:project-detail", "lookup_field": "slug"}
         }
 
+    def create(self, validated_data):
+        source_language_validated = validated_data.get("source_language")
+        if source_language_validated:
+            source_language = Language.objects.get(
+                code=source_language_validated.get("code")
+            )
+            if source_language:
+                validated_data["source_language"] = source_language
```

Comment:
```suggestion
            validated_data["source_language"] = Language.objects.get(
                code=source_language_validated.get("code")
            )
```

The get will return object or thrown an exception, so the condition here is not really useful.